frame/system/src/lib.rs

```diff
@@ -711,13 +419,354 @@ decl_module! {
+		/// `:code` was updated.
+		CodeUpdated,
+		/// A new \[account\] was created.
+		NewAccount(T::AccountId),
+		/// An \[account\] was reaped.
+		KilledAccount(T::AccountId),
+	}
+
+	/// Old name generated by `decl_event`.
+	#[deprecated(note = "use `Event` instead")]
+	pub type RawEvent<T> = Event<T>;
+
+	/// Error for the System module
+	#[pallet::error]
+	pub enum Error<T> {
+		/// The name of specification does not match between the current runtime
+		/// and the new runtime.
+		InvalidSpecName,
+		/// The specification version is not allowed to decrease between the current runtime
+		/// and the new runtime.
+		SpecVersionNeedsToIncrease,
+		/// Failed to extract the runtime version from the new runtime.
+		///
+		/// Either calling `Core_version` or decoding `RuntimeVersion` failed.
+		FailedToExtractRuntimeVersion,
+		/// Suicide called when the account has non-default composite data.
+		NonDefaultComposite,
+		/// There is a non-zero reference count preventing the account from being purged.
+		NonZeroRefCount,
+	}
+
+	/// Exposed trait-generic origin type.
+	#[pallet::origin]
+	pub type Origin<T> = RawOrigin<<T as Config>::AccountId>;
+
+	/// The full account information for a particular account ID.
+	#[pallet::storage]
+	#[pallet::getter(fn account)]
+	pub type Account<T: Config> = StorageMap<
+		_,
+		Blake2_128Concat,
+		T::AccountId,
+		AccountInfo<T::Index, T::AccountData>,
+		ValueQuery,
+	>;
+
+	/// Total extrinsics count for the current block.
+	#[pallet::storage]
+	pub(super) type ExtrinsicCount<T: Config> = StorageValue<_, u32>;
+
+	/// The current weight for the block.
+	#[pallet::storage]
+	#[pallet::getter(fn block_weight)]
+	pub(super) type BlockWeight<T: Config> = StorageValue<_, ConsumedWeight, ValueQuery>;
+
+	/// Total length (in bytes) for all extrinsics put together, for the current block.
+	#[pallet::storage]
+	pub(super) type AllExtrinsicsLen<T: Config> = StorageValue<_, u32>;
+
+	/// Map of block numbers to block hashes.
+	#[pallet::storage]
+	#[pallet::getter(fn block_hash)]
+	pub type BlockHash<T: Config> =
+		StorageMap<_, Twox64Concat, T::BlockNumber, T::Hash, ValueQuery>;
+
+	/// Extrinsics data for the current block (maps an extrinsic's index to its data).
+	#[pallet::storage]
+	#[pallet::getter(fn extrinsic_data)]
+	pub(super) type ExtrinsicData<T: Config> =
+		StorageMap<_, Twox64Concat, u32, Vec<u8>, ValueQuery>;
+
+	/// The current block number being processed. Set by `execute_block`.
+	#[pallet::storage]
+	#[pallet::getter(fn block_number)]
+	pub(super) type Number<T: Config> = StorageValue<_, T::BlockNumber, ValueQuery>;
+
+	/// Hash of the previous block.
+	#[pallet::storage]
+	#[pallet::getter(fn parent_hash)]
+	pub(super) type ParentHash<T: Config> = StorageValue<_, T::Hash, ValueQuery>;
+
+	/// Digest of the current block, also part of the block header.
+	#[pallet::storage]
+	#[pallet::getter(fn digest)]
+	pub(super) type Digest<T: Config> = StorageValue<_, DigestOf<T>, ValueQuery>;
```

Comment:
Why do we need all those `pub(super)`? Can't we already access it via `#[pallet::generate_store(pub (super) trait Store)`?

Reply:
I think if the type is not `pub(super)` the trait `Store` cannot be `pub(super)`, otherwise compilation fail because it cannot expose private types.

Reply:
@thiolliere? All the storage items were generated via the template.

Reply:
> I think if the type is not `pub(super)` the trait `Store` cannot be `pub(super)`, otherwise compilation fail because it cannot expose private types.

And this is because the macro creates `__PrefixForDigest` struct with same visibility as `Digest` type, and use it for the first generic `_`. Otherwise it would have worked: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=c1c8b33e024018e443d27e76ac573a82